### ISSUE TITLE
False positive assigning-non-slot when __slots__ defined on class inheriting from Generic[...]

### ISSUE THREAD

**Issue by msg555** [PyCQA/pylint#2939]
Tested on pylint 2.3.1

### Steps to reproduce

Consider the below module

```
""" Demonstrate pylint generics __slots__ bug """
from typing import (
    Generic,
    TypeVar,
)

T = TypeVar('T')

class Foo(Generic[T]):
    """ Simple class with slots """
    __slots__ = ['value']

    def __init__(self, value):
        self.value = value
```

### Current behavior

pylint gives the below error

`[E0237(assigning-non-slot), Foo.__init__] Assigning to attribute 'value' not defined in class slots`

### Expected behavior

No error should be detected.

### pylint --version output

```
pylint 2.3.1
astroid 2.2.5
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0]
```
